pkg/asset/metadata/metadata.go

```diff
@@ -0,0 +1,80 @@
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+
+	// "github.com/hashicorp/terraform/terraform"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+)
+
+const (
+	// MetadataFilename is name of the file where clustermetadata is stored.
+	MetadataFilename  = "metadata.json"
+	metadataAssetName = "Cluster Metadata"
+)
+
+// Metadata depends on cluster and installconfig,
+type Metadata struct {
+	installConfig asset.Asset
+	cluster       asset.Asset
+}
+
+var _ asset.Asset = (*Metadata)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (m *Metadata) Name() string {
+	return metadataAssetName
+}
+
+// Dependencies returns the dependency of the MetaData.
+func (m *Metadata) Dependencies() []asset.Asset {
+	return []asset.Asset{m.installConfig, m.cluster}
+}
+
+// Generate generates the metadata.yaml file.
+func (m *Metadata) Generate(parents map[asset.Asset]*asset.State) (*asset.State, error) {
+	installCfg, err := installconfig.GetInstallConfig(m.installConfig, parents)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get install config state in the parent asset states")
```

Comment:
Can't we just return `err` here?  It's [already][1] both specific about what went wrong and (hopefully) has enough context to stand alone.  If you feel the existing error are lacking, can you update `GetInstallConfig` so all consumers can benefit from the improved error messages?

[1]: https://github.com/openshift/installer/blob/238a0eda9c197eac642d2163c5ab9de210ff2179/pkg/asset/installconfig/installconfig.go#L158-L162

Reply:
done. returns `err`